### PR TITLE
Update privacy notice form

### PR DESCRIPTION
**Pull request by jpople**
Closes PROD-1457

### Description Of Changes

Updates privacy notice form and table to work with new endpoints.  Note that the component used for selecting languages here is temporary and should be changed to use the new ScrollableList component-- ticket for that change is [here](https://ethyca.atlassian.net/browse/PROD-1770?atlOrigin=eyJpIjoiOTY5ZjM1Nzc5NTYwNDRmZThlNjY0NmFjNGNmZmQ0MmUiLCJwIjoiaiJ9).

### Code Changes

* [ ] Privacy notice form is reworked to accommodate model changes
* [ ] Add new PrivacyNoticeTranslationForm component
* [ ] Notice endpoints that were previously batch changes have been changed to single notice endpoints with id
* [ ] Privacy notice table has been updated to use the new `limited_update` endpoint that's specific to enable/disable notices

### Steps to Confirm

* [ ] Navigate to `http://localhost:3000/consent/privacy-notices`
* [ ] Add a new privacy notice with translations, save
* [ ] Edit an existing privacy notice, save
* [ ] Add a system with an applicable data use to one of your privacy notices
* [ ] Return to notices table, enable/disable corresponding notice

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/admin-ui/src/features/common/Layout.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ const Layout = ({
     <Flex
       data-testid={title}
       direction="column"
-      height={padded ? "calc(100vh - 48px" : "100vh"}
+      height={padded ? "calc(100vh - 48px)" : "100vh"}
```
Comment (eastandwestwind) [inline clients/admin-ui/src/features/common/Layout.tsx:57]: ⭐

File: clients/admin-ui/src/features/privacy-notices/PrivacyNoticeForm.tsx [hunk 2]
```diff
@@ -67,10 +59,16 @@ const PrivacyNoticeForm = ({
...
+      const valuesToSubmit = {
+        ...values,
+        id: passedInPrivacyNotice!.id,
+        translations: values.translations ?? [],
+      };
+      result = await patchNoticesMutationTrigger(valuesToSubmit);
     } else {
-      result = await postNoticesMutationTrigger([values]);
+      // // eslint-disable-next-line
+      // const { notice_key, ...valuesToSubmit } = values;
```
Comment (eastandwestwind) [inline clients/admin-ui/src/features/privacy-notices/PrivacyNoticeForm.tsx:70]: can we remove this commented-out code?

**Review by eastandwestwind — CHANGES_REQUESTED** [review]
Code looks solid, but had some functional issues while testing the language list. See https://www.loom.com/share/9bbce69d11b3458890b81a0efd21a39b

**Comment by eastandwestwind** [comment]
Functionally looking great now, aside from this issue when attempting to enable a notice:

Screenshot 2024-03-01 at 4 52 30 PM

**Review by eastandwestwind — APPROVED** [review]

**Comment by jpople** [comment]
Confirmed with @Kelsey-Ethyca that enable/disable is functioning OK, merging.

**Pull request merged**